File: src/main/java/no/nav/sosialhjelp/soknad/web/rest/mappers/VedleggTypeToSoknadTypeMapper.java
//package no.nav.sosialhjelp.soknad.web.rest.mappers;
//
//import java.util.HashMap;
//import java.util.Map;
//
//import static no.nav.sbl.soknadsosialhjelp.json.SoknadJsonTyper.BARNEBIDRAG;
//import static no.nav.sbl.soknadsosialhjelp.json.SoknadJsonTyper.FORMUE_ANNET;
//import static no.nav.sbl.soknadsosialhjelp.json.SoknadJsonTyper.FORMUE_BRUKSKONTO;
//import static no.nav.sbl.soknadsosialhjelp.json.SoknadJsonTyper.FORMUE_BSU;
//import static no.nav.sbl.soknadsosialhjelp.json.SoknadJsonTyper.FORMUE_LIVSFORSIKRING;
//import static no.nav.sbl.soknadsosialhjelp.json.SoknadJsonTyper.FORMUE_SPAREKONTO;
//import static no.nav.sbl.soknadsosialhjelp.json.SoknadJsonTyper.FORMUE_VERDIPAPIRER;
//import static no.nav.sbl.soknadsosialhjelp.json.SoknadJsonTyper.JOBB;
//import static no.nav.sbl.soknadsosialhjelp.json.SoknadJsonTyper.SLUTTOPPGJOER;
//import static no.nav.sbl.soknadsosialhjelp.json.SoknadJsonTyper.STUDIELAN;
//import static no.nav.sbl.soknadsosialhjelp.json.SoknadJsonTyper.UTBETALING_ANNET;
//import static no.nav.sbl.soknadsosialhjelp.json.SoknadJsonTyper.UTBETALING_FORSIKRING;
//import static no.nav.sbl.soknadsosialhjelp.json.SoknadJsonTyper.UTBETALING_HUSBANKEN;
//import static no.nav.sbl.soknadsosialhjelp.json.SoknadJsonTyper.UTBETALING_SALG;
//import static no.nav.sbl.soknadsosialhjelp.json.SoknadJsonTyper.UTBETALING_UTBYTTE;
//import static no.nav.sbl.soknadsosialhjelp.json.SoknadJsonTyper.UTGIFTER_ANDRE_UTGIFTER;
//import static no.nav.sbl.soknadsosialhjelp.json.SoknadJsonTyper.UTGIFTER_ANNET_BARN;
//import static no.nav.sbl.soknadsosialhjelp.json.SoknadJsonTyper.UTGIFTER_ANNET_BO;
//import static no.nav.sbl.soknadsosialhjelp.json.SoknadJsonTyper.UTGIFTER_BARNEHAGE;
//import static no.nav.sbl.soknadsosialhjelp.json.SoknadJsonTyper.UTGIFTER_BARN_FRITIDSAKTIVITETER;
//import static no.nav.sbl.soknadsosialhjelp.json.SoknadJsonTyper.UTGIFTER_BARN_TANNREGULERING;
//import static no.nav.sbl.soknadsosialhjelp.json.SoknadJsonTyper.UTGIFTER_BOLIGLAN_AVDRAG;
//import static no.nav.sbl.soknadsosialhjelp.json.SoknadJsonTyper.UTGIFTER_HUSLEIE;
//import static no.nav.sbl.soknadsosialhjelp.json.SoknadJsonTyper.UTGIFTER_KOMMUNAL_AVGIFT;
//import static no.nav.sbl.soknadsosialhjelp.json.SoknadJsonTyper.UTGIFTER_OPPVARMING;
//import static no.nav.sbl.soknadsosialhjelp.json.SoknadJsonTyper.UTGIFTER_SFO;
//import static no.nav.sbl.soknadsosialhjelp.json.SoknadJsonTyper.UTGIFTER_STROM;
//import static no.nav.sbl.soknadsosialhjelp.json.SoknadJsonTyper.VERDI_ANNET;
//import static no.nav.sbl.soknadsosialhjelp.json.SoknadJsonTyper.VERDI_BOLIG;
//import static no.nav.sbl.soknadsosialhjelp.json.SoknadJsonTyper.VERDI_CAMPINGVOGN;
//import static no.nav.sbl.soknadsosialhjelp.json.SoknadJsonTyper.VERDI_FRITIDSEIENDOM;
//import static no.nav.sbl.soknadsosialhjelp.json.SoknadJsonTyper.VERDI_KJORETOY;
//
//public final class VedleggTypeToSoknadTypeMapper {
//
//    private VedleggTypeToSoknadTypeMapper() {
//    }
//
//    public static final Map<String,String> vedleggTypeToSoknadType = new HashMap<>();
//
//    static {
//        vedleggTypeToSoknadType.put("kontooversikt|aksjer", FORMUE_VERDIPAPIRER);
//        vedleggTypeToSoknadType.put("faktura|annetbarnutgift", UTGIFTER_ANNET_BARN);
//        vedleggTypeToSoknadType.put("dokumentasjon|annetboutgift", UTGIFTER_ANNET_BO);
//        vedleggTypeToSoknadType.put("dokumentasjon|annetinntekter", UTBETALING_ANNET);
//        vedleggTypeToSoknadType.put("dokumentasjon|annetverdi", VERDI_ANNET); // Økonomisk verdi. Usikker på om denne brukes
//        vedleggTypeToSoknadType.put("dokumentasjon|campingvogn", VERDI_CAMPINGVOGN); // Økonomisk verdi. Usikker på om denne brukes
//        vedleggTypeToSoknadType.put("dokumentasjon|fritidseiendom", VERDI_FRITIDSEIENDOM); // Økonomisk verdi. Usikker på om denne brukes
//        vedleggTypeToSoknadType.put("kjopekontrakt|kjopekontrakt", VERDI_BOLIG); // Økonomisk verdi. Usikker på om denne brukes
//        vedleggTypeToSoknadType.put("dokumentasjon|kjoretoy", VERDI_KJORETOY); // Økonomisk verdi. Usikker på om denne brukes
//        vedleggTypeToSoknadType.put("faktura|barnehage", UTGIFTER_BARNEHAGE);
//        vedleggTypeToSoknadType.put("barnebidrag|betaler", BARNEBIDRAG);
//        vedleggTypeToSoknadType.put("kontooversikt|brukskonto", FORMUE_BRUKSKONTO);
//        vedleggTypeToSoknadType.put("kontooversikt|bsu", FORMUE_BSU);
//        vedleggTypeToSoknadType.put("salgsoppgjor|eiendom", UTBETALING_SALG);
//        vedleggTypeToSoknadType.put("dokumentasjon|forsikringsutbetaling", UTBETALING_FORSIKRING);
//        vedleggTypeToSoknadType.put("faktura|fritidsaktivitet", UTGIFTER_BARN_FRITIDSAKTIVITETER);
//        vedleggTypeToSoknadType.put("faktura|husleie", UTGIFTER_HUSLEIE);
//        vedleggTypeToSoknadType.put("faktura|kommunaleavgifter", UTGIFTER_KOMMUNAL_AVGIFT);
//        vedleggTypeToSoknadType.put("kontooversikt|livsforsikring", FORMUE_LIVSFORSIKRING);
//        vedleggTypeToSoknadType.put("barnebidrag|mottar", BARNEBIDRAG);
//        vedleggTypeToSoknadType.put("faktura|oppvarming", UTGIFTER_OPPVARMING);
//        vedleggTypeToSoknadType.put("faktura|sfo", UTGIFTER_SFO);
//        vedleggTypeToSoknadType.put("kontooversikt|sparekonto", FORMUE_SPAREKONTO);
//        vedleggTypeToSoknadType.put("faktura|strom", UTGIFTER_STROM);
//        vedleggTypeToSoknadType.put("faktura|tannbehandling", UTGIFTER_BARN_TANNREGULERING);
//        vedleggTypeToSoknadType.put("dokumentasjon|utbytte", UTBETALING_UTBYTTE);
//        vedleggTypeToSoknadType.put("husbanken|vedtak", UTBETALING_HUSBANKEN);
//        vedleggTypeToSoknadType.put("student|vedtak", STUDIELAN);
//        vedleggTypeToSoknadType.put("lonnslipp|arbeid", JOBB);
//        vedleggTypeToSoknadType.put("sluttoppgjor|arbeid", SLUTTOPPGJOER);
//        vedleggTypeToSoknadType.put("kontooversikt|annet", FORMUE_ANNET);
//        vedleggTypeToSoknadType.put("annet|annet", UTGIFTER_ANDRE_UTGIFTER);
//        vedleggTypeToSoknadType.put("dokumentasjon|annet", UTBETALING_ANNET);
//        vedleggTypeToSoknadType.put("nedbetalingsplan|avdraglaan", UTGIFTER_BOLIGLAN_AVDRAG); // vedleggstypen er også knyttet til soknadstypen "boliglanRenter"
//    }
//
//    public static String getSoknadPath(String vedleggType){
//        switch (vedleggType){
//            case "dokumentasjon|annetboutgift":
//            case "faktura|annetbarnutgift":
//            case "faktura|tannbehandling":
//            case "faktura|kommunaleavgifter":
//            case "faktura|fritidsaktivitet":
//            case "faktura|oppvarming":
//            case "faktura|strom":
//            case "annet|annet":
//                return "opplysningerUtgift";
//            case "barnebidrag|betaler":
//            case "faktura|sfo":
//            case "faktura|barnehage":
//            case "faktura|husleie":
//            case "nedbetalingsplan|avdraglaan":
//                return "oversiktUtgift";
//            case "dokumentasjon|kjoretoy":
//            case "dokumentasjon|campingvogn":
//            case "dokumentasjon|fritidseiendom":
//            case "dokumentasjon|annetverdi":
//            case "kjopekontrakt|kjopekontrakt":
//            case "kontooversikt|brukskonto":
//            case "kontooversikt|bsu":
//            case "kontooversikt|sparekonto":
//            case "kontooversikt|livsforsikring":
//            case "kontooversikt|aksjer":
//            case "kontooversikt|annet":
//                return "formue";
//            case "dokumentasjon|forsikringsutbetaling":
//            case "dokumentasjon|annetinntekter":
//            case "dokumentasjon|utbytte":
//            case "dokumentasjon|annet":
//            case "salgsoppgjor|eiendom":
//            case "sluttoppgjor|arbeid":
//            case "husbanken|vedtak":
//                return "utbetaling";
//            case "barnebidrag|mottar":
//            case "lonnslipp|arbeid":
//            case "student|vedtak":
//                return "inntekt";
//            default:
//                throw new IllegalStateException("Vedleggstypen eksisterer ikke eller mangler mapping");
//        }
//    }
//
//    public static boolean isInSoknadJson(String vedleggType) {
//        switch (vedleggType){
//            case "oppholdstillatel|oppholdstillatel":
//            case "samvarsavtale|barn":
//            case "husleiekontrakt|husleiekontrakt":
//            case "husleiekontrakt|kommunal":
//            case "skattemelding|skattemelding":
//                return false;
//            default:
//                return true;
//        }
//    }
//}

File: src/main/kotlin/no/nav/sosialhjelp/soknad/okonomiskeopplysninger/mappers/VedleggTypeToSoknadTypeMapper.kt
package no.nav.sosialhjelp.soknad.okonomiskeopplysninger.mappers

import no.nav.sbl.soknadsosialhjelp.json.SoknadJsonTyper
import no.nav.sosialhjelp.soknad.okonomiskeopplysninger.dto.VedleggType
import no.nav.sosialhjelp.soknad.okonomiskeopplysninger.dto.VedleggType.AnnetAnnet
import no.nav.sosialhjelp.soknad.okonomiskeopplysninger.dto.VedleggType.BarnebidragBetaler
import no.nav.sosialhjelp.soknad.okonomiskeopplysninger.dto.VedleggType.BarnebidragMottar
import no.nav.sosialhjelp.soknad.okonomiskeopplysninger.dto.VedleggType.DokumentasjonAnnetBoutgift
import no.nav.sosialhjelp.soknad.okonomiskeopplysninger.dto.VedleggType.DokumentasjonAnnetInntekter
import no.nav.sosialhjelp.soknad.okonomiskeopplysninger.dto.VedleggType.DokumentasjonForsikringsutbetaling
import no.nav.sosialhjelp.soknad.okonomiskeopplysninger.dto.VedleggType.DokumentasjonUtbytte
import no.nav.sosialhjelp.soknad.okonomiskeopplysninger.dto.VedleggType.FakturaAnnetBarnutgift
import no.nav.sosialhjelp.soknad.okonomiskeopplysninger.dto.VedleggType.FakturaBarnehage
import no.nav.sosialhjelp.soknad.okonomiskeopplysninger.dto.VedleggType.FakturaFritidsaktivitet
import no.nav.sosialhjelp.soknad.okonomiskeopplysninger.dto.VedleggType.FakturaHusleie
import no.nav.sosialhjelp.soknad.okonomiskeopplysninger.dto.VedleggType.FakturaKommunaleavgifter
import no.nav.sosialhjelp.soknad.okonomiskeopplysninger.dto.VedleggType.FakturaOppvarming
import no.nav.sosialhjelp.soknad.okonomiskeopplysninger.dto.VedleggType.FakturaSfo
import no.nav.sosialhjelp.soknad.okonomiskeopplysninger.dto.VedleggType.FakturaStrom
import no.nav.sosialhjelp.soknad.okonomiskeopplysninger.dto.VedleggType.FakturaTannbehandling
import no.nav.sosialhjelp.soknad.okonomiskeopplysninger.dto.VedleggType.HusbankenVedtak
import no.nav.sosialhjelp.soknad.okonomiskeopplysninger.dto.VedleggType.KontooversiktAksjer
import no.nav.sosialhjelp.soknad.okonomiskeopplysninger.dto.VedleggType.KontooversiktAnnet
import no.nav.sosialhjelp.soknad.okonomiskeopplysninger.dto.VedleggType.KontooversiktBrukskonto
import no.nav.sosialhjelp.soknad.okonomiskeopplysninger.dto.VedleggType.KontooversiktBsu
import no.nav.sosialhjelp.soknad.okonomiskeopplysninger.dto.VedleggType.KontooversiktLivsforsikring
import no.nav.sosialhjelp.soknad.okonomiskeopplysninger.dto.VedleggType.KontooversiktSparekonto
import no.nav.sosialhjelp.soknad.okonomiskeopplysninger.dto.VedleggType.LonnslippArbeid
import no.nav.sosialhjelp.soknad.okonomiskeopplysninger.dto.VedleggType.NedbetalingsplanAvdragslan
import no.nav.sosialhjelp.soknad.okonomiskeopplysninger.dto.VedleggType.SalgsoppgjorEiendom
import no.nav.sosialhjelp.soknad.okonomiskeopplysninger.dto.VedleggType.SluttoppgjorArbeid
import no.nav.sosialhjelp.soknad.okonomiskeopplysninger.dto.VedleggType.StudentVedtak

object VedleggTypeToSoknadTypeMapper {
    val vedleggTypeToSoknadType: Map<VedleggType, String> =
        mapOf(
            AnnetAnnet to SoknadJsonTyper.UTGIFTER_ANDRE_UTGIFTER,
            BarnebidragBetaler to SoknadJsonTyper.BARNEBIDRAG,
            BarnebidragMottar to SoknadJsonTyper.BARNEBIDRAG,
            DokumentasjonAnnetBoutgift to SoknadJsonTyper.UTGIFTER_ANNET_BO,
            DokumentasjonAnnetInntekter to SoknadJsonTyper.UTBETALING_ANNET,
            DokumentasjonUtbytte to SoknadJsonTyper.UTBETALING_UTBYTTE,
            DokumentasjonForsikringsutbetaling to SoknadJsonTyper.UTBETALING_FORSIKRING,
            FakturaBarnehage to SoknadJsonTyper.UTGIFTER_BARNEHAGE,
            FakturaAnnetBarnutgift to SoknadJsonTyper.UTGIFTER_ANNET_BARN,
            FakturaFritidsaktivitet to SoknadJsonTyper.UTGIFTER_BARN_FRITIDSAKTIVITETER,
            FakturaTannbehandling to SoknadJsonTyper.UTGIFTER_BARN_TANNREGULERING,
            FakturaStrom to SoknadJsonTyper.UTGIFTER_STROM,
            FakturaOppvarming to SoknadJsonTyper.UTGIFTER_OPPVARMING,
            FakturaSfo to SoknadJsonTyper.UTGIFTER_SFO,
            FakturaKommunaleavgifter to SoknadJsonTyper.UTGIFTER_KOMMUNAL_AVGIFT,
            FakturaHusleie to SoknadJsonTyper.UTGIFTER_HUSLEIE,
            HusbankenVedtak to SoknadJsonTyper.UTBETALING_HUSBANKEN,
            KontooversiktLivsforsikring to SoknadJsonTyper.FORMUE_LIVSFORSIKRING,
            KontooversiktAnnet to SoknadJsonTyper.FORMUE_ANNET,
            KontooversiktBrukskonto to SoknadJsonTyper.FORMUE_BRUKSKONTO,
            KontooversiktAksjer to SoknadJsonTyper.FORMUE_VERDIPAPIRER,
            KontooversiktSparekonto to SoknadJsonTyper.FORMUE_SPAREKONTO,
            KontooversiktBsu to SoknadJsonTyper.FORMUE_BSU,
            LonnslippArbeid to SoknadJsonTyper.JOBB,
            NedbetalingsplanAvdragslan to SoknadJsonTyper.UTGIFTER_BOLIGLAN_AVDRAG,
            SalgsoppgjorEiendom to SoknadJsonTyper.UTBETALING_SALG,
            StudentVedtak to SoknadJsonTyper.STUDIELAN,
            SluttoppgjorArbeid to SoknadJsonTyper.SLUTTOPPGJOER,
            // vedleggstypen er også knyttet til soknadstypen "boliglanRenter"
        )

    fun getSoknadPath(vedleggType: VedleggType?): String =
        when (vedleggType) {
            DokumentasjonAnnetBoutgift, FakturaAnnetBarnutgift, FakturaTannbehandling, FakturaKommunaleavgifter,
            FakturaFritidsaktivitet, FakturaOppvarming, FakturaStrom, AnnetAnnet,
            -> "opplysningerUtgift"
            BarnebidragBetaler, FakturaSfo, FakturaBarnehage, FakturaHusleie, NedbetalingsplanAvdragslan -> "oversiktUtgift"
            KontooversiktBrukskonto, KontooversiktBsu, KontooversiktSparekonto, KontooversiktLivsforsikring,
            KontooversiktAksjer, KontooversiktAnnet,
            -> "formue"
            DokumentasjonForsikringsutbetaling, DokumentasjonAnnetInntekter, DokumentasjonUtbytte, SalgsoppgjorEiendom,
            SluttoppgjorArbeid, HusbankenVedtak,
            -> "utbetaling"
            BarnebidragMottar, LonnslippArbeid, StudentVedtak -> "inntekt"
            else -> error("Vedleggstypen eksisterer ikke eller mangler mapping")
        }
}
